web_src/less/_repository.less

```diff
@@ -1336,110 +1369,29 @@
   .timeline-item.commits-list .singular-commit .sha.label {
     border: 1px solid var(--color-light-border);
 
-    .ui.signature.avatar {
-      height: 16px;
-      margin-bottom: 0;
-      width: 16px;
+    .sha-label-colors(@color) {
+      --color-light-border: @color;
+      --color-light: fade(@color, 10%);
+      --color-hover: fade(@color, 30%);
```

Comment:
Can these vars be moved to `:root`? Generally I prefer to define vars only on root and not on selectors.

Reply:
These are already defined in root. These are used in the same way theming works, in that it overrides the colors for these elements so the inherited rules could source these colors.

Before this PR, multiple rules are defined for each and every class. This simplifies a lot, even in the compiled CSS code (as `--color-light-border` is used twice).

Reply:
I think we should not use functions like `fade` because those will not be changeable via CSS vars at runtime which I would still like to maintain. So instead of using `fade`, please pre-calculate the resulting colors and create/update the vars at `:root`.

Reply:
> I would still like to maintain to allow theming based purely on CSS vars alone.

This has already made them fully changeable by custom themes - one would just override them at runtime at their subclass level. I see no reason why so many variables need to go into `:root`. These rules are already a lot more manageable to me.

I mean, writing
```
: root {
  --color-sha-label-color: 
  --color-sha-label-color-background:
  --color-sha-label-color-hover: 
  --color-sha-label-verified-color:
  --color-sha-label-verified-color-background:
  --color-sha-label-verified-color-hover:
[...]
}
```
or
```
.sha.label.isSigned {
  --color-light-border:
  --color-light:
  --color-hover:
}
.sha.label.isSignedVerified {
  --color-light-border:
  --color-light:
  --color-hover:
}
[...]
```
doesn't make *that* much difference IMHO...

Reply:
With that method, you're tieing the theme to the selectors, e.g. if we ever change the `.sha.label.isSigned` selector, it's going to break themes, as opposed to just tieing the theme to just the variable name on `:root` which is a much better interface imho.

Reply:
Okay, but I suggest we spread this kind of variables into top level selectors, like `.repository {}` instead of `:root`, but still let them live in `_base.less` to make them easier to find. I just hate seeing a bunch of variables defined in root in dev tools. Searching through these is very painful due to how browsers group and filter rules.

Similarly, `--color-input-text:` & co. could become `.input { --color-text: }` as they are of lower specificity than the actual `.ui.input`. Putting everything into `:root` is like going back in time before classes were born...

Reply:
Going through the code, I actually think it's valid to just use `.input { --color-text: }`, though I can't say this for all selectors. In doubt, put `--color-text` into `body { --color-text }` as well, or use something like `--g-color-text`...

Reply:
> Putting everything into :root is like going back in time before classes were born

It is the most simple way and I think it's generally what is recommended and many sites like GitHub do exactly this. If you check Github, they have like 500 rules in `:root` and they are very easy to theme because all vars live on one selector. I think it's the way to go.